### ISSUE TITLE
Fulltext-Index creation/dropping should not throw Exceptions

### ISSUE THREAD

**Issue by jastax** [neo4j/neo4j#12287]
Neo4j 3.5.8

Calling CALL db.index.fulltext.createNodeIndex twice results in an Error. 

![grafik](https://user-images.githubusercontent.com/1672178/64235862-88deeb80-cef9-11e9-9cf9-890a485f4088.png)

The same thing happens when calling CALL db.index.fulltext.drop('native_search_summary') twice or when the index does not exists.

![grafik](https://user-images.githubusercontent.com/1672178/64236095-fdb22580-cef9-11e9-946b-d69a10f3676e.png)

This is a problem when deploying an application. When it's deployed the first time, createNodeIndex would have to be called. Deploying it the second time the application would now throw an Exception. In this case, the index exists and  db.index.fulltext.drop would have to be called first. So the new order would be

drop
createNodeIndex

Deploying this application the first time to a fresh database would thus throw an Exception.

This was not a problem with apoc.index.remove and apoc.index.addAllNodes if I remember correctly.

There also seems to be a problem with mixing indexes
![grafik](https://user-images.githubusercontent.com/1672178/64236477-c98b3480-cefa-11e9-84fa-1cb8504c16df.png)

The old index could be created/updated multiple times (with or without different names)

![grafik](https://user-images.githubusercontent.com/1672178/64236627-140cb100-cefb-11e9-8b75-e493d3729e37.png)

**Comment by burqen** [comment]
Hello @jastax 
You raise two problems:
1. CALL db.index.fulltext.createNodeIndex twice with the same arguments results in an Error.
2. Not able to create db.index.fulltext on same label - property combination as existing apoc index.

1. This is the intended behaviour for schema indexes. You can work around it by checking for existence before creating the index. `CALL db.indexes()` will be useful for that.
2. I'm not able to reproduce this issue the way you've described it. Can you please go into more details on how you reproduce the issue?

**Comment by jastax** [comment]
Hello @burqen, 

thanks for your reply.

This is certainly a design decision that can be made, but it is inconsistent with earlier behavior and other indexes:

![grafik](https://user-images.githubusercontent.com/1672178/64608956-1d4ebf80-d3cc-11e9-9e4f-fd0b1c6058aa.png)

If you prefer to keep that design as-is, how about adding a db.index.fulltext.createNodeIndexIfNotExists?

i.e. POSTGRES (9.5+) allows CREATE INDEX IF NOT EXISTS

Regarding the second issue, I think I had another index created (other than native_search_summary) as I can't reproduce the issue either. Maybe it should be allowed to define multiple indexes using the same nodes/properties though.

**Comment by burqen** [comment]
@jastax Indeed it is. Looking closer at this I need to revise my answer:

> This is the ~intended~ **current** behaviour for ~schema indexes~ **fulltext indexes**. You can work around it by checking for existence before creating the index. CALL db.indexes() will be useful for that.

Fulltext indexes are currently only half way integrated with the product as a whole. The only way of interacting with is through the fulltext procedures. We are working towards having fulltext indexes more tightly integrated and this inconsistency in behaviour should be addressed in that piece of work. The intention is for fulltext indexes to behave the same way as "normal" indexes. So thank you again for pointing this out. 

Allowing multiple indexes on the same node-property combination (but different index name) is also a thing that we are working towards.

Unfortunately I don't have any estimate for when this work will be done.

**Comment by Hunterness** [comment]
So an update from the Cypher team:
* We updated the Cypher command for index creation in 4.0 (to introduce naming indexes, name is optional and will get generated if left out) which also included starting to throw errors for creating the same index twice: `CREATE INDEX index_name FOR (n:Label) ON (n.prop)`
* We also introduced a new drop command to drop by the name which throws an error when dropping non-existing indexes: `DROP INDEX index_name`
* In 4.1 we added an optional flag to not throw these errors: `CREATE INDEX index_name IF NOT EXISTS FOR (n:Label) ON (n.prop)` and `DROP INDEX index_name IF EXISTS`
* In 4.2 we introduced a Cypher command for listing indexes, `SHOW INDEXES` which replaces the procedure `db.indexes`. This will be enhanced in 4.3 to allow `YIELD` and `WHERE` clauses to filter the result better.
* In 4.3 we will add Cypher commands for creating/dropping fulltext indexes as to no longer rely on procedures for this (they will still be queried by procedures). An example of creating a fulltext index is `CREATE FULLTEXT INDEX fulltext_index_name FOR (n:Label1|Label2) ON EACH [n.prop1, n.prop2]` which also allows the `IF NOT EXISTS` flag. Dropping is done with the same command as the other indexes, `DROP INDEX fulltext_index_name` which allows the `IF EXISTS` flag.

I hope this helps you :)
/Therese, Team Cypher
